pints/toy/_constant_model.py

```diff
@@ -0,0 +1,64 @@
+#
+# Constant model with multiple outputs.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2018, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import numpy as np
+import pints
+
+
+class ConstantModel(pints.ForwardModel):
```

Comment:
If it's not too cumbersome, could we change this model so that you have to define the number of parameters and outputs when you create it? I'd like this to be constant throughout the model's lifetime, and not change everything you make a call to it. That's going to cause trouble somewhere down the line

Reply:
So I did think (and as I said in the commit) actually try and implement this. The problem I was having with this, though, is that if, whenever we instantiate the model, we declare the parameters, this causes issues downstream with simulate. 

Imagine,

params = [0]
model = ConstantModel(params)

Then later, I want to simulate the model over a set of times (using the same set of parameters),

values = model.simulate(times)

That's fine. But what I can't do is use that same model to simulate a new set of data with new parameters because in Python you can't have an optional argument before a obligatory one. So I can't define a 'simulate' function as: simulate(self, parameters=None, times), meaning that I'd have to change the order of parameters and times and make this model different to all other models.

How about I make this model the same as all others by instantiating it with nothing.

def __init__(self):
        super(ConstantModel, self).__init__()

Then self._parameters is only set when simulate is called, like all other functions. Seem ok? That's what the latest commit does.

Reply:
I think you should still pass it parameters. It should just ignore them (since the output is constant)